autospec/pkg_integrity.py

```diff
@@ -210,32 +206,43 @@ def __del__(self):
         print(SEPT)
 
 
-def head_request(url):
-    curl = pycurl.Curl()
-    curl.setopt(curl.URL, url)
-    curl.setopt(curl.CUSTOMREQUEST, "HEAD")
-    curl.setopt(curl.NOBODY, True)
-    curl.setopt(curl.TIMEOUT, 5)
-    curl.setopt(curl.FOLLOWLOCATION, True)
-    curl.perform()
-    http_code = curl.getinfo(pycurl.HTTP_CODE)
-    curl.close()
-    return http_code
+def download_file(url, destination):
+    sign_file = os.path.join(destination, os.path.basename(url))
+    print("Attempting {}".format(sign_file))
+    with open(sign_file, 'wb') as f:
+        curl = pycurl.Curl()
+        curl.setopt(curl.URL, url)
+        curl.setopt(curl.WRITEDATA, f)
+        curl.setopt(curl.FOLLOWLOCATION, True)
+        try:
+            curl.perform()
+        except pycurl.error as e:
+            print(e.args)
+            return None
+        code = curl.getinfo(pycurl.HTTP_CODE)
+        curl.close()
+        if code != 200:
+            os.unlink(sign_file)
+            return None
+        print("Downloaded {}!".format(sign_file))
```

Comment:
I don't think this message needs the exclamation mark

Reply:
Sorry, couple of prints are leftovers.